iree/libraries/CMakeLists.txt

```diff
@@ -0,0 +1,113 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This package defines "mondo" static and shared libraries corresponding
+# to various slices of IREE's codebase (intended for external consumption).
+# Generally, libraries are aggregated per-directory.
+#
+# Note that if depending on these from CMake (i.e. for demos/samples/etc),
+# then it is important to not *also* link against the individual libraries
+# in the main source tree, as doing so will produce duplicate symbols. These
+# libraries represent the public interface for third parties to depend on
+# IREE. When using static libraries from outside of CMake, users must take
+# care to link against these libraries and their dependencies in the correct
+# order. As an example:
+#
+#   -liree_runtime_core_drivers -liree_runtime_base -lflatcc -lcpuinfo -ldl
+#
+# See the DEPS sections of each.
+#
+# Also, when used from within CMake, compile definitions (include directories,
+# defines, etc) are automatically managed. From the outside, these must
+# be specified.
+
+################################################################################
+# iree_runtime_base library
+# Contains the core runtime and supporting code. Does not contain any drivers.
+################################################################################
+
+iree_aggregate_library(
+  NAME
+    runtime_base
+  OUTPUT_NAME
+    iree_runtime_base
+  PACKAGES
+    iree_base
+    iree_base_internal
+    iree_hal
```

Comment:
> Since it gets this we should be able to get the same listing from cmake of dependent libs. If not, all of these are iree_cc_library and just like you added stuff here we can add whatever we need to make that work. @ScottTodd actually had this working ages ago for alwayslink and we may be able to revive that.

Just starting a thread here to drop some references to compare with.

* https://github.com/google/iree/pull/205
* https://github.com/google/iree/pull/483